packages/rich-text/src/plugins/Table/index.tsx

```diff
@@ -199,11 +207,66 @@ function hasHeadersOutsideFirstRow(nodes: CustomElement[]) {
     .some(({ children }) => (children as CustomElement[]).some(isTableHeaderCell));
 }
 
+const isAtEndOfTextSelection = (editor: SPEditor) =>
+  editor.selection?.focus.offset === getText(editor, editor.selection?.focus.path).length;
+
+function currentSelectionStartsTableCell(editor: SPEditor): boolean {
+  const [tableCellNode, path] = getNodeEntryFromSelection(editor, [
+    BLOCKS.TABLE_CELL,
+    BLOCKS.TABLE_HEADER_CELL,
+  ]);
+  return (tableCellNode && !getText(editor, path)) || editor.selection?.focus.offset === 0;
+}
+
+/**
+ * This traversal strategy is unfortunately necessary because Slate doesn't
+ * expose something like Node.next(editor).
+ */
+function getNextNode(editor: SPEditor): CustomElement | null {
```

Comment:
Can we move this and perhaps `isAtEndOfTextSelection` to a utils/helper file? They seem pretty generic and a bit distracting from the tables logic in here.